monai/networks/nets/ahnet.py

```diff
@@ -0,0 +1,515 @@
+
+        return out
+
+
+class Projection(nn.Sequential):
+    def __init__(self, num_input_features: int, num_output_features: int):
+        super(Projection, self).__init__()
+
+        conv3d_type: Type[nn.Conv3d] = Conv[Conv.CONV, 3]
+        norm3d_type: Type[nn.BatchNorm3d] = Norm[Norm.BATCH, 3]
+        relu_type: Type[nn.ReLU] = Act[Act.RELU]
+
+        self.add_module("norm", norm3d_type(num_input_features))
+        self.add_module("relu", relu_type(inplace=True))
+        self.add_module("conv", nn.Conv3d(num_input_features, num_output_features, kernel_size=1, stride=1, bias=False))
+
+
+class DenseBlock(nn.Sequential):
+    def __init__(self, num_layers: int, num_input_features: int, bn_size: int, growth_rate: int, drop_rate: float):
+        super(DenseBlock, self).__init__()
+        for i in range(num_layers):
+            layer = Pseudo3DLayer(num_input_features + i * growth_rate, growth_rate, bn_size, drop_rate)
+            self.add_module("denselayer%d" % (i + 1), layer)
+
+
+class UpTransition(nn.Sequential):
+    def __init__(self, num_input_features: int, num_output_features: int, deter_flag: bool = False):
+        super(UpTransition, self).__init__()
+
+        conv3d_type: Type[nn.Conv3d] = Conv[Conv.CONV, 3]
+        norm3d_type: Type[nn.BatchNorm3d] = Norm[Norm.BATCH, 3]
+        relu_type: Type[nn.ReLU] = Act[Act.RELU]
+
+        self.add_module("norm", norm3d_type(num_input_features))
+        self.add_module("relu", relu_type(inplace=True))
+        self.add_module(
+            "conv", conv3d_type(num_input_features, num_output_features, kernel_size=1, stride=1, bias=False)
+        )
+        if deter_flag:
+            conv3d_trans_type: Type[nn.ConvTranspose3d] = Conv[Conv.CONVTRANS, 3]
+            self.add_module(
+                "pool", conv3d_trans_type(num_output_features, num_output_features, kernel_size=2, stride=2, bias=False)
+            )
+        else:
+            self.add_module("pool", nn.Upsample(scale_factor=2, mode="trilinear", align_corners=True))
+
+
+class Final(nn.Sequential):
+    def __init__(self, num_input_features: int, num_output_features: int, deter_flag: bool = False):
+        super(Final, self).__init__()
+
+        conv3d_type: Type[nn.Conv3d] = Conv[Conv.CONV, 3]
+        norm3d_type: Type[nn.BatchNorm3d] = Norm[Norm.BATCH, 3]
+        relu_type: Type[nn.ReLU] = Act[Act.RELU]
+
+        self.add_module("norm", norm3d_type(num_input_features))
+        self.add_module("relu", relu_type(inplace=True))
+        self.add_module(
+            "conv",
+            conv3d_type(
+                num_input_features, num_output_features, kernel_size=(3, 3, 1), stride=1, padding=(1, 1, 0), bias=False
+            ),
+        )
+        if deter_flag:
+            conv3d_trans_type: Type[nn.ConvTranspose3d] = Conv[Conv.CONVTRANS, 3]
+            self.add_module(
+                "pool", conv3d_trans_type(num_output_features, num_output_features, kernel_size=2, stride=2, bias=False)
```

Comment:
upsampling is named 'pool'? is this a typo?

Reply:
It comes from the author's official code: see https://github.com/lsqshr/AH-Net/blob/master/net3d.py  (line 195), can I change it into "up"?

Reply:
sure thanks